src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -204,6 +204,13 @@ void BackgroundNotificationCb (NSNotification obj)
 		void RemoveInflightData (NSUrlSessionTask task, bool cancel = true)
 		{
 			lock (inflightRequestsLock) {
+				InflightData data;
+				if (inflightRequests.TryGetValue (task, out data)) {
+					data = inflightRequests [task];
+					if (cancel)
+						data.CancellationTokenSource.Cancel ();
```

Comment:
Needed to ensure that when the original cancellation token is cancelled, we do cancel the one in the streams.